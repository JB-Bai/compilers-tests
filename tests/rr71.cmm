float;
struct qE_VZySW{
}
QXSD68JK4;
float vspBCvfZBP(){
	int tFLxPxyJQ = CSJTVoO;
	float djT39u = sQ6HSspQca[U9LVRzM(-(0252135) / - ! - OWQpA3_muW = 0x4B[PSPTKVaL].lQrA <= 062.8577e+1[S3], ! 545.53.npizB, dQeBNY)];
	int z8P9Z, r5B[0x2b][067];
	struct BsGEn58rJ pfjIC_t[0063][0xC6];
	q4pgyWd3;
	if(uSb8sLmCdJc((SaoKrGY + IFnyW3 && 31.429) *(! ! - 0x71E1 + yxW4Nku(07310123, 00200 / 820.516e+1) / 0xd25a * 6.955e+0.GU8WJGkq6z))) 040526;
	-(flffrm381() + 053.03e0)[k0eoAW] && QypViHJF || 532.444e+3 || Pnaz() = 341.3136 / HIH1tWaa() != 0x8 / EAo627GDdR != 03421 / 79.199e-5.YdF7EA7 + UJYtq8sLzQ() * 57.716E9 == 00.0E-8 / 026453 + 23006 - ! 0x8E60 && - 0x838 && whNaY1S;
	return 64.60 >= UQeb2r(2.2123);
	0xF36b = 26.3 / - rzNklE7AFjK;
	0x8Acf;
	if(62.227E9) SekTqd() - vFdgfIaGo = 0xFAc + 0xd2Bb > IltRLJMFF && 4430[gviyOB[153.384.yLTcvjy_]] = ! lu3PWm = 0x8f6;
	else VWw5lpj();
}

