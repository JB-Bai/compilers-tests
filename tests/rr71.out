Program (1)
  ExtDefList (1)
    ExtDef (1)
      Specifier (1)
        TYPE: float
      SEMI
    ExtDefList (2)
      ExtDef (2)
        Specifier (2)
          StructSpecifier (2)
            STRUCT
            OptTag (2)
              ID: qE_VZySW
            LC
            RC
        ExtDecList (4)
          VarDec (4)
            ID: QXSD68JK4
        SEMI
      ExtDefList (5)
        ExtDef (5)
          Specifier (5)
            TYPE: float
          FunDec (5)
            ID: vspBCvfZBP
            LP
            RP
          CompSt (5)
            LC
            DefList (6)
              Def (6)
                Specifier (6)
                  TYPE: int
                DecList (6)
                  Dec (6)
                    VarDec (6)
                      ID: tFLxPxyJQ
                    ASSIGNOP
                    Exp (6)
                      ID: CSJTVoO
                SEMI
              DefList (7)
                Def (7)
                  Specifier (7)
                    TYPE: float
                  DecList (7)
                    Dec (7)
                      VarDec (7)
                        ID: djT39u
                      ASSIGNOP
                      Exp (7)
                        Exp (7)
                          ID: sQ6HSspQca
                        LB
                        Exp (7)
                          ID: U9LVRzM
                          LP
                          Args (7)
                            Exp (7)
                              Exp (7)
                                Exp (7)
                                  MINUS
                                  Exp (7)
                                    LP
                                    Exp (7)
                                      INT: 87133
                                    RP
                                DIV
                                Exp (7)
                                  MINUS
                                  Exp (7)
                                    NOT
                                    Exp (7)
                                      MINUS
                                      Exp (7)
                                        ID: OWQpA3_muW
                              ASSIGNOP
                              Exp (7)
                                Exp (7)
                                  Exp (7)
                                    Exp (7)
                                      INT: 75
                                    LB
                                    Exp (7)
                                      ID: PSPTKVaL
                                    RB
                                  DOT
                                  ID: lQrA
                                RELOP
                                Exp (7)
                                  Exp (7)
                                    FLOAT: 628.577026
                                  LB
                                  Exp (7)
                                    ID: S3
                                  RB
                            COMMA
                            Args (7)
                              Exp (7)
                                NOT
                                Exp (7)
                                  Exp (7)
                                    FLOAT: 545.530029
                                  DOT
                                  ID: npizB
                              COMMA
                              Args (7)
                                Exp (7)
                                  ID: dQeBNY
                          RP
                        RB
                  SEMI
                DefList (8)
                  Def (8)
                    Specifier (8)
                      TYPE: int
                    DecList (8)
                      Dec (8)
                        VarDec (8)
                          ID: z8P9Z
                      COMMA
                      DecList (8)
                        Dec (8)
                          VarDec (8)
                            VarDec (8)
                              VarDec (8)
                                ID: r5B
                              LB
                              INT: 43
                              RB
                            LB
                            INT: 55
                            RB
                    SEMI
                  DefList (9)
                    Def (9)
                      Specifier (9)
                        StructSpecifier (9)
                          STRUCT
                          Tag (9)
                            ID: BsGEn58rJ
                      DecList (9)
                        Dec (9)
                          VarDec (9)
                            VarDec (9)
                              VarDec (9)
                                ID: pfjIC_t
                              LB
                              INT: 51
                              RB
                            LB
                            INT: 198
                            RB
                      SEMI
            StmtList (10)
              Stmt (10)
                Exp (10)
                  ID: q4pgyWd3
                SEMI
              StmtList (11)
                Stmt (11)
                  IF
                  LP
                  Exp (11)
                    ID: uSb8sLmCdJc
                    LP
                    Args (11)
                      Exp (11)
                        Exp (11)
                          LP
                          Exp (11)
                            Exp (11)
                              Exp (11)
                                ID: SaoKrGY
                              PLUS
                              Exp (11)
                                ID: IFnyW3
                            AND
                            Exp (11)
                              FLOAT: 31.429001
                          RP
                        STAR
                        Exp (11)
                          LP
                          Exp (11)
                            Exp (11)
                              NOT
                              Exp (11)
                                NOT
                                Exp (11)
                                  MINUS
                                  Exp (11)
                                    INT: 29153
                            PLUS
                            Exp (11)
                              Exp (11)
                                Exp (11)
                                  ID: yxW4Nku
                                  LP
                                  Args (11)
                                    Exp (11)
                                      INT: 1937491
                                    COMMA
                                    Args (11)
                                      Exp (11)
                                        Exp (11)
                                          INT: 128
                                        DIV
                                        Exp (11)
                                          FLOAT: 8205.160156
                                  RP
                                DIV
                                Exp (11)
                                  INT: 53850
                              STAR
                              Exp (11)
                                Exp (11)
                                  FLOAT: 6.955000
                                DOT
                                ID: GU8WJGkq6z
                          RP
                    RP
                  RP
                  Stmt (11)
                    Exp (11)
                      INT: 16726
                    SEMI
                StmtList (12)
                  Stmt (12)
                    Exp (12)
                      Exp (12)
                        Exp (12)
                          Exp (12)
                            Exp (12)
                              MINUS
                              Exp (12)
                                Exp (12)
                                  LP
                                  Exp (12)
                                    Exp (12)
                                      ID: flffrm381
                                      LP
                                      RP
                                    PLUS
                                    Exp (12)
                                      FLOAT: 53.029999
                                  RP
                                LB
                                Exp (12)
                                  ID: k0eoAW
                                RB
                            AND
                            Exp (12)
                              ID: QypViHJF
                          OR
                          Exp (12)
                            FLOAT: 532444.000000
                        OR
                        Exp (12)
                          ID: Pnaz
                          LP
                          RP
                      ASSIGNOP
                      Exp (12)
                        Exp (12)
                          Exp (12)
                            Exp (12)
                              Exp (12)
                                Exp (12)
                                  Exp (12)
                                    FLOAT: 341.313599
                                  DIV
                                  Exp (12)
                                    ID: HIH1tWaa
                                    LP
                                    RP
                                RELOP
                                Exp (12)
                                  Exp (12)
                                    INT: 8
                                  DIV
                                  Exp (12)
                                    ID: EAo627GDdR
                              RELOP
                              Exp (12)
                                Exp (12)
                                  Exp (12)
                                    INT: 1809
                                  DIV
                                  Exp (12)
                                    Exp (12)
                                      FLOAT: 0.000792
                                    DOT
                                    ID: YdF7EA7
                                PLUS
                                Exp (12)
                                  Exp (12)
                                    ID: UJYtq8sLzQ
                                    LP
                                    RP
                                  STAR
                                  Exp (12)
                                    FLOAT: 57715998720.000000
                            RELOP
                            Exp (12)
                              Exp (12)
                                Exp (12)
                                  Exp (12)
                                    FLOAT: 0.000000
                                  DIV
                                  Exp (12)
                                    INT: 11563
                                PLUS
                                Exp (12)
                                  INT: 23006
                              MINUS
                              Exp (12)
                                NOT
                                Exp (12)
                                  INT: 36448
                          AND
                          Exp (12)
                            MINUS
                            Exp (12)
                              INT: 2104
                        AND
                        Exp (12)
                          ID: whNaY1S
                    SEMI
                  StmtList (13)
                    Stmt (13)
                      RETURN
                      Exp (13)
                        Exp (13)
                          FLOAT: 64.599998
                        RELOP
                        Exp (13)
                          ID: UQeb2r
                          LP
                          Args (13)
                            Exp (13)
                              FLOAT: 2.212300
                          RP
                      SEMI
                    StmtList (14)
                      Stmt (14)
                        Exp (14)
                          Exp (14)
                            INT: 62315
                          ASSIGNOP
                          Exp (14)
                            Exp (14)
                              FLOAT: 26.299999
                            DIV
                            Exp (14)
                              MINUS
                              Exp (14)
                                ID: rzNklE7AFjK
                        SEMI
                      StmtList (15)
                        Stmt (15)
                          Exp (15)
                            INT: 35535
                          SEMI
                        StmtList (16)
                          Stmt (16)
                            IF
                            LP
                            Exp (16)
                              FLOAT: 62227001344.000000
                            RP
                            Stmt (16)
                              Exp (16)
                                Exp (16)
                                  Exp (16)
                                    ID: SekTqd
                                    LP
                                    RP
                                  MINUS
                                  Exp (16)
                                    ID: vFdgfIaGo
                                ASSIGNOP
                                Exp (16)
                                  Exp (16)
                                    Exp (16)
                                      Exp (16)
                                        Exp (16)
                                          INT: 4012
                                        PLUS
                                        Exp (16)
                                          INT: 53947
                                      RELOP
                                      Exp (16)
                                        ID: IltRLJMFF
                                    AND
                                    Exp (16)
                                      Exp (16)
                                        INT: 4430
                                      LB
                                      Exp (16)
                                        Exp (16)
                                          ID: gviyOB
                                        LB
                                        Exp (16)
                                          Exp (16)
                                            FLOAT: 153.384003
                                          DOT
                                          ID: yLTcvjy_
                                        RB
                                      RB
                                  ASSIGNOP
                                  Exp (16)
                                    Exp (16)
                                      NOT
                                      Exp (16)
                                        ID: lu3PWm
                                    ASSIGNOP
                                    Exp (16)
                                      INT: 2294
                              SEMI
                            ELSE
                            Stmt (17)
                              Exp (17)
                                ID: VWw5lpj
                                LP
                                RP
                              SEMI
            RC
